tests/kernel/device/kernel.device.pm (and tests/subsys/pm/power_mgmt/subsys.pm.device_pm) fails to build on mec172xevb_assy6906 & mec1501modular_assy6885

`./scripts/twister -s tests/kernel/device/kernel.device.pm -p mec1501modular_assy6885 -p mec172xevb_assy6906`

gets:

```
[ 32%] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/pinmux/pinmux_mchp_xec.c.obj
In file included from /home/galak/git/zephyr/include/kernel_includes.h:21,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/include/init.h:11,
                 from /home/galak/git/zephyr/include/device.h:29,
                 from /home/galak/git/zephyr/drivers/pinmux/pinmux_mchp_xec.c:8:
/home/galak/git/zephyr/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device.pm/zephyr/include/generated/devicetree_unfixed.h:1697:56: error: 'DT_N_S_pinmux_S_pinmux_0_P_wakeup_source' undeclared here (not in a function); did you mean 'DT_N_S_cpus_S_cpu_0_P_wakeup_source'?
 1697 | #define DT_N_S_pinmux_FOREACH_CHILD_STATUS_OKAY(fn) fn(DT_N_S_pinmux_S_pinmux_0) fn(DT_N_S_pinmux_S_pinmux_1) fn(DT_N_S_pinmux_S_pinmux_2) fn(DT_N_S_pinmux_S_pinmux_3) fn(DT_N_S_pinmux_S_pinmux_4) fn(DT_N_S_pinmux_S_pinmux_5)
```

--- a/include/device.h
+++ b/include/device.h
@@ -732,13 +732,17 @@ BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 		(&DEVICE_NAME_GET(dev_name)), level, prio)
 
 #ifdef CONFIG_PM_DEVICE
+/* Use of DT_PROP_OR here is because we cant assume that 'wakeup-source`
+ * will be a defined property for the binding of the devicetree node that
+ * is associated with the device
+ */
 #define Z_DEVICE_STATE_DEFINE(node_id, dev_name)			\
 	static struct device_state Z_DEVICE_STATE_NAME(dev_name) = {	\
 		.pm = {						        \
 			.flags = ATOMIC_INIT(COND_CODE_1(		\
 					DT_NODE_EXISTS(node_id),	\
-					(DT_PROP(			\
-					node_id, wakeup_source)),	\
+					(DT_PROP_OR(			\
+					node_id, wakeup_source, 0)),	\
 					(0)) <<			        \
 				PM_DEVICE_FLAGS_WS_CAPABLE),		\
 		},                                                      \

